pkg/handlers/generic/mutation/imageregistries/credentials/credentials_secret.go

```diff
@@ -114,5 +114,5 @@ func kubeletStaticCredentialProviderSecretContents(configs []providerConfig) (st
 }
 
 func credentialSecretName(clusterName string) string {
-	return fmt.Sprintf("%s-registry-creds", clusterName)
+	return fmt.Sprintf("%s-credential-provider-response", clusterName)
```

Comment:
I was a little concerned about the length of this (even before the change tbh) but I looked into it and all is good here. Secret name is maximum 253 characters ([DNS subdomain name](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go#L282)) and Cluster name is maximum 63 characters ([label name](https://github.com/kubernetes-sigs/cluster-api/blob/main/internal/webhooks/cluster.go#L167)).